src/main/java/evaluation/ParameterSearch.java

```diff
@@ -0,0 +1,210 @@
+package evaluation;
+
+import core.AbstractPlayer;
+import evodef.*;
+import games.GameType;
+import ntbea.*;
+import players.PlayerFactory;
+import players.mcts.MCTSSearchSpace;
+import players.simple.RandomPlayer;
+import utilities.Pair;
+import utilities.StatSummary;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static utilities.Utils.getArg;
+
+public class ParameterSearch {
+
+    public static void main(String[] args) {
+        List<String> argsList = Arrays.asList(args);
+        if (argsList.contains("--help") || argsList.contains("-h")) System.out.println(
+                "The first three arguments must be \n" +
+                        "\t<filename for searchSpace definition>\n" +
+                        "\t<number of NTBEA iterations>\n" +
+                        "\t<game type> \n" +
+                        "Then there are a number of optional arguments:\n" +
+                        "\tnPlayers=      The total number of players in each game (the default is game.Min#players) \n " +
+                        "\tevalGames=     The number of games to run with the best predicted setting to estimate its true value (default is 20% of NTBEA iterations) \n" +
+                        "\topponent=      The filename for the agent used as opponent. Default is a Random player. \n" +
+                        "\tuseThreeTuples If specified then we use 3-tuples as well as 1-, 2- and N-tuples \n" +
+                        "\tkExplore=      The k to use in NTBEA - defaults to 100.0 \n" +
+                        "\thood=          The size of neighbourhood to look at in NTBEA. Default is min(50, |searchSpace|/100) \n" +
+                        "\trepeat=        The number of times NTBEA should be re-run, to find a single best recommendation \n" +
+                        "\tverbose        Will log the results marginalised to each dimension, and the Top 10 best tuples for each run \n"
+        );
+
+        if (argsList.size() < 3)
+            throw new AssertionError("Must specify at least three parameters: searchSpace, NTBEA iterations, game");
+        String searchSpaceFile = args[0];
+        int iterationsPerRun = Integer.parseInt(args[1]);
+        GameType game = GameType.valueOf(args[2]);
+//        if (game != GameType.Dominion)
+//            throw new AssertionError("Only Dominion currently supported");
+        int repeats = getArg(args, "repeat", 1);
+        int evalGames = getArg(args, "evalGames", iterationsPerRun / 5);
+        double kExplore = getArg(args, "kExplore", 100.0);
+        String opponentFile = getArg(args, "opponent", "");
+        boolean verbose = Arrays.asList(args).contains("verbose");
+        int nPlayers = getArg(args, "nPlayers", game.getMinPlayers());
+
+        MCTSSearchSpace searchSpace = new MCTSSearchSpace(searchSpaceFile);
```

Comment:
Something to consider here is that games also have parameter classes that can be tuned. So changing the concepts above of "opponents" to "players" and adding parsing of games parameters as well. Although I'm not sure how much extra work / classes it'd involve to add all the games in this way, maybe reworking ITunableParameters and having a different version working with that format would be enough, keeping this separate for agent optimisation.

Reply:
In principle, a big "yes". Any objections if we leave that to a separate Issue/Pull Request?

I've raised an issue already to properly integrate ITunableParameters with the NTBEA libraries.